src/sentry/db/models/fields/encrypted.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import absolute_import
+
+__all__ = (
+    'EncryptedCharField',
+    'EncryptedJsonField',
+    'EncryptedPickledObjectField',
+    'EncryptedTextField',
+)
+
+import six
+
+from django.db.models import CharField, TextField
+from jsonfield import JSONField
+from picklefield.fields import PickledObjectField
+from sentry.utils.encryption import decrypt, encrypt
+
+
+class EncryptedCharField(CharField):
+    def get_db_prep_value(self, value, *args, **kwargs):
+        value = super(EncryptedCharField, self).get_db_prep_value(
+            value, *args, **kwargs)
+        return encrypt(value)
+
+    def to_python(self, value):
+        if value is not None and isinstance(value, six.string_types):
+            value = decrypt(value)
+        return super(EncryptedCharField, self).to_python(value)
+
+    def get_prep_lookup(self, lookup_type, value):
+        raise NotImplementedError('{!r} lookup type for {!r} is not supported'.format(
+            lookup_type,
+            self,
+        ))
+
+
+class EncryptedJsonField(JSONField):
+    def get_db_prep_value(self, value, *args, **kwargs):
+        value = super(EncryptedJsonField, self).get_db_prep_value(
+            value, *args, **kwargs)
+        return encrypt(value)
+
+    def to_python(self, value):
+        if value is not None and isinstance(value, six.string_types):
+            value = decrypt(value)
+        return super(EncryptedJsonField, self).to_python(value)
+
+    def get_prep_lookup(self, lookup_type, value):
+        raise NotImplementedError('{!r} lookup type for {!r} is not supported'.format(
+            lookup_type,
+            self,
+        ))
+
+
+class EncryptedPickledObjectField(PickledObjectField):
+    def get_db_prep_value(self, value, *args, **kwargs):
+        if isinstance(value, six.binary_type):
```

Comment:
This seems odd to me that we're coercing all binary values into utf-8 strings.

Same on the `to_python` side, we're explicitly want `string_type` spit out instead of raw binary. I feel like I'd expect the exact opposite. Raw bytes in, raw bytes out. Up to the caller to handle encoding/decoding.

Then these fields can subclass `BinaryField` iirc.